File: src/main/java/net/slipp/service/qna/MigrationService.java
package net.slipp.service.qna;

import java.util.List;

import javax.annotation.Resource;

import net.slipp.domain.qna.Answer;
import net.slipp.domain.qna.Question;
import net.slipp.repository.qna.AnswerRepository;
import net.slipp.repository.qna.QuestionRepository;

import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@Transactional
public class MigrationService {
	@Resource(name = "questionRepository")
	private QuestionRepository questionRepository;
	
	@Resource(name = "answerRepository")
	private AnswerRepository answerRepository;
	
	public void convertConfluenceToMarkdown() {
		List<Question> questions = questionRepository.findAll();
		for (Question question : questions) {
			question.convertWiki();
		}
		
		List<Answer> answers = answerRepository.findAll();
		for (Answer answer : answers) {
			answer.convertWiki();
		}
	}
}

File: src/main/scala/net/slipp/service/qna/MigrationService.scala
package net.slipp.service.qna

import java.util.List
import javax.annotation.Resource

import net.slipp.domain.qna.Question
import net.slipp.repository.qna.{AnswerRepository, QuestionRepository}
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

import scala.collection.JavaConversions._

@Service
@Transactional
class MigrationService {
  @Resource(name = "questionRepository") private var questionRepository: QuestionRepository = _
  @Resource(name = "answerRepository") private var answerRepository: AnswerRepository = _

  def convertConfluenceToMarkdown {
    val questions: List[Question] = questionRepository.findAll
    questions.foreach(q => q.convertWiki())
    answerRepository.findAll.foreach(a => a.convertWiki())
  }

  def migrateFacebookPostId {
    val questions = questionRepository.findByDeleted(false);
    questions.foreach(question => {
      question.migrateFacebookPostId
    });
  }
}